Src/AutoFixtureUnitTest/FreezeOnMatchCustomizationTest.cs

```diff
@@ -77,6 +90,63 @@ public void InitializeWithMatcherShouldSetCorrespondingProperty()
             Assert.Equal(matcher, sut.Matcher);
         }
 
+
+        [Fact]
+        public void InitializeWithSingleNullRequestArgumentShouldThrowArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null));
+        }
+
+        [Fact]
+        public void InitializeWithNullRequestShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null, new FalseRequestSpecification()));
+            Assert.Equal("request", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithRequestAndNullMatcherShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                  new FreezeOnMatchCustomization(new object(), null));
+            Assert.Equal("matcher", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithSingleRequestShouldSetCorrespondingProperty()
+        {
+            var request = new object();
+            var sut = new FreezeOnMatchCustomization(request);
+            Assert.Same(request, sut.Request);
+        }
+
+        [Fact]
+        public void InitializeWithObjectRequestShouldSetTrueRequestSpecificationMatcher()
```

Comment:
Well, I'm fine with structural inspection as soon as we agree that it's fine to not follow the [append-only](http://blog.ploeh.dk/2013/04/02/why-trust-tests/) approach with such tests. Recently I had another PR where I [had to alter](https://github.com/AutoFixture/AutoFixture/pull/781/files#diff-68dade75f99c180341e9e21ce0f0d3be) all such tests because they are pretty volatile.

Currently we have the following block of code:
```csharp
var targetType = request as Type;
if (targetType != null)
{
    this.Matcher = new ExactTypeSpecification(targetType);
}
else
{
    this.Matcher = new EqualRequestSpecification(request);
}
```
This code is redundant and can be simplified to the `EqualRequestSpecification` only, but you [don't want](https://github.com/AutoFixture/AutoFixture/pull/697#issuecomment-243843127) to simplify it because of [this](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixtureUnitTest/FreezeOnMatchCustomizationTest.cs#L65) test which is a pure structural inspection. 

In such light I'm pretty reluctant to add this test in which you pin that specification should be necessarily of the `EqualRequestSpecification` type. If later we'll need to tune behavior even more (e.g. add some conditions), we will be limited by this test.

If we agree that tests should be append only (even structural inspection), then I'd vote to remove this test. If we agree that structural inspection tests could be altered, let's simplify the code block above and adjust [this](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixtureUnitTest/FreezeOnMatchCustomizationTest.cs#L65) test.

@sergeyshushlyapin What do you think about this? 😉
@moodmosaic Do you have any opinion about the [structural inspection](http://enterprisecraftsmanship.com/2016/07/21/unit-testing-anti-patterns-structural-inspection/) tests? Do you agree that it's fine to change them if necessary like I did [here](https://github.com/AutoFixture/AutoFixture/pull/781/files#diff-68dade75f99c180341e9e21ce0f0d3be)?